rust-runtime/aws-smithy-runtime-api/src/client/runtime_plugin.rs

```diff
@@ -235,21 +235,13 @@ impl RuntimePlugins {
         Default::default()
     }
 
-    pub fn with_client_plugin(mut self, plugin: impl RuntimePlugin + 'static) -> Self {
-        insert_plugin!(
-            self.client_plugins,
-            plugin,
-            SharedRuntimePlugin::new(plugin)
-        );
+    pub fn with_client_plugin(mut self, plugin: SharedRuntimePlugin) -> Self {
```

Comment:
if we end up keeping this change we might want to update the changelog. I think there is already some customer code using this.

Reply:
Should be fine after 01720a953b12fc775e3f0c37d83d6f6887bba4b8